operator/controllers/csiscaleoperator_controller.go

```diff
@@ -1553,3 +1562,55 @@ func (r *CSIScaleOperatorReconciler) resourceExists(instance *csiscaleoperator.C
 		return true, nil
 	}
 }
+
+// getConfigMap fetches data from the "ibm-spectrum-scale-csi-config" configmap from the cluster
+// and returns a configmap reference.
+func (r *CSIScaleOperatorReconciler) getConfigMap(instance *csiscaleoperator.CSIScaleOperator, name string) (*corev1.ConfigMap, error) {
+
+	logger := csiLog.WithName("getCSIConfig").WithValues("Kind", corev1.ResourceConfigMaps, "Name", name)
+	logger.Info("Read CSI configmap from the cluster.")
+
+	cm := &corev1.ConfigMap{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      name,
+		Namespace: instance.Namespace,
+	}, cm)
+
+	if err != nil && errors.IsNotFound(err) {
+		message := "Resource not found."
+		logger.Info(message)
+		//return nil instead of error to not to keep reconciling, just proceed with other things
+		//when configmap is not found.
+		return cm, nil
+	} else if err != nil {
+		message := "Failed to get resource information from cluster."
+		logger.Error(err, message)
+		// TODO: Add event.
+		meta.SetStatusCondition(&crStatus.Conditions, metav1.Condition{
+			Type:    string(config.StatusConditionSuccess),
+			Status:  metav1.ConditionFalse,
+			Reason:  string(csiv1.ResourceReadError),
+			Message: message,
+		})
+		return cm, err
+	} else {
+		return cm, nil
+	}
+}
+
+// parseConfigMap parses the data in the configMap in the desired format(VAR_DRIVER_ENV_NAME: VALUE to ENV_NAME: VALUE).
+func parseConfigMap(cm *corev1.ConfigMap, resource string) map[string]string {
+
+	logger := csiLog.WithName("parseConfigMap").WithValues("Name", cm.Name)
+	logger.Info("Parsing the data from configmap.", "configmap", cm.Name)
+
+	data := map[string]string{}
+	if resource == config.CSINode.String() {
+		for key, value := range cm.Data {
+			if strings.HasPrefix(strings.ToUpper(key), "VAR_DRIVER") && key[10] == '_' {
```

Comment:
A condition `strings.HasPrefix(strings.ToUpper(key), "VAR_DRIVER_")` should be sufficient, right? any specific reason to have 2 separate conditions?